http/client/src/commonMain/kotlin/org/jetbrains/packagesearch/api/v3/http/PackageSearchApiClient.kt

```diff
@@ -211,40 +399,54 @@ public class PackageSearchApiClient(
     public suspend fun searchProjects(
         request: SearchProjectRequest,
         requestBuilder: (HttpRequestBuilder.() -> Unit)?,
-    ): List<ApiProject> =
-        defaultRequest<_, List<ApiProject>>(
+    ): List<ApiProject> {
+
+        val apiProjectsCache = apiProjectCacheCollection.await()
+
+        apiProjectsCache.find(selector = SearchCacheEntry::key.name, value = request).firstOrNull()
+            ?.also { if (!it.isExpired()) return it.value }
+
+        return defaultRequest<_, List<ApiProject>>(
             method = HttpMethod.Post,
             url = endpoints.searchPackages,
             body = request,
             requestBuilder = requestBuilder,
-        )
+        ).also {
+            apiProjectsCache.updateWhere(
+                fieldSelector = ApiProjectCacheEntry::key.name,
+                fieldValue = request,
+                upsert = true,
+                update = ApiProjectCacheEntry(request, it)
+            )
+        }
```

Comment:
Save also the packages here, just like in `startScroll`.

Reply:
ApiProject contains only the ids of packages, so we can't

Reply:
Ah yes, my bad.